backends/authority/person.go

```diff
@@ -0,0 +1,245 @@
+package authority
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"slices"
+
+	"github.com/elastic/go-elasticsearch/v6/esapi"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/ugent-library/biblio-backoffice/models"
+)
+
+//go:embed person_settings.json
+var personSettings string
+
+func (c *Client) EnsurePersonSeedIndexExists() error {
+	res, err := esapi.IndicesExistsRequest{
+		Index: []string{"biblio_person"},
+	}.Do(context.Background(), c.es)
+	if err != nil {
+		return err
+	}
+	if res.StatusCode == 404 {
+		res, err := c.es.Indices.Create("biblio_person", c.es.Indices.Create.WithBody(strings.NewReader(personSettings)))
+		if err != nil {
+			return err
+		}
+		if res.IsError() {
+			return fmt.Errorf("%+v", res)
+		}
+	}
+	return nil
+}
+
+func (c *Client) SeedPerson(data []byte) error {
```

Comment:
Why do you provide a byte array instead of a Person object?

Reply:
they are not serializations Person objects, but the structure that the frontoffice expects